sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -308,12 +308,15 @@ private class ScriptTransformationWriterThread(
       }
       threwException = false
     } catch {
+      // SPARK-25158 Exception should not be thrown again, otherwise it will be captured by
```

Comment:
All exceptions? How about the fatal exceptions? Should we do it only when it is NonFatal?

Reply:
All exceptions already raise to `ScriptTransformationExec` use `_exception`, they're all dealt with `catch` block of `TaskRunner` whatever it is a NonFatal Exception or a Fatal Exception